app/javascript/publishers/dashboard_chart.js

```diff
@@ -0,0 +1,105 @@
+import Chart from 'chart.js';
+import moment from 'moment';
+
+let colors = [
+  '255, 99, 132',
+  '54, 162, 235',
+  '255, 206, 86',
+  '75, 192, 192',
+  '153, 102, 255',
+  '255, 159, 64'
+]
+
+function createLabels(startingDate) {
+  // https://stackoverflow.com/questions/7556591/is-the-javascript-date-object-always-one-day-off
+  var loop = new Date(startingDate.replace(/-/g, '\/'));
+  var dates_array = [];
+
+  while (loop <= new Date()) {
+    dates_array.push(loop.getFullYear() + '-' + (loop.getMonth() + 1) + '-' + loop.getDate());
+    loop.setDate(loop.getDate() + 1);
+  }
+
+  return dates_array;
+}
+
+// Max of the chart is 80% of the suggested max to be used by Chartjs
+function getSuggestedMax(data) {
+  var currentMax = 0;
+  Object.keys(data).forEach(function (key) {
+    var value = data[key];
+    currentMax = value.retrievals > currentMax ? value.retrievals : currentMax;
+    currentMax = value.first_runs > currentMax ? value.first_runs : currentMax;
+    currentMax = value.finalized > currentMax ? value.finalized : currentMax;
+  });
+  return (currentMax * 100 / 95)
+}
+
+function createCharts() {
+  var data;
+  JSON.parse(document.getElementById('referrals-hidden-tags').value).forEach(function (element) {
+    data = JSON.parse(document.getElementById(element).value);
+    createChart(
+      data,
+      element.replace(/-/g, " "),
+      getSuggestedMax(data)
+    );
+  });
+}
+
+function createChart(data, title, suggestedMax) {
+  var wrapper = document.getElementById('channel-referrals-stats-chart');
+  var canvas = document.createElement('canvas');
+  canvas.setAttribute("width", "400");
+  canvas.setAttribute("height", "300");
+  wrapper.appendChild(canvas);
+
+  Chart.defaults.global.defaultFontFamily = 'Poppins';
+
+  new Chart(canvas, {
+    type: 'line',
+    data: {
+      labels: createLabels(data[0]['ymd']),
+      datasets: [
+        {
+          label: 'Retrievals',
```

Comment:
Let's call this Downloads to be consistent with the referral promo. (I'll make an item to fix the referral promo  box naming, it should actually be installs).